src/lsm/k_way_merge_benchmark.zig

```diff
@@ -0,0 +1,174 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+const stdx = @import("stdx");
+
+const Bench = @import("../testing/bench.zig");
+const Pending = error{Pending};
+const KWayMergeIteratorType = @import("k_way_merge.zig").KWayMergeIteratorType;
+
+const streams_count_max = 32;
+const repetitions: usize = 32;
+
+const body_fmt = "K={:_>2} L={:_>3} WT={}";
+
+// Those Values are close to the real-world use case.
+const Values = .{
+    ValueType(u64, 128),
+    ValueType(u256, 32),
+    ValueType(u256, 32),
+    ValueType(u256, 32),
+    ValueType(u128, 16),
+    ValueType(u64, 8),
+    ValueType(u128, 16),
+};
+
+test "benchmark: k-way-merge" {
+    var bench: Bench = .init();
+    defer bench.deinit();
+
+    bench.report("WT: Wall time/merged element", .{});
+
+    const streams_count: usize = @intCast(bench.parameter("streams_count", 4, 32));
+    const stream_length: usize = @intCast(bench.parameter("stream_length", 128, 8192));
+    assert(streams_count <= streams_count_max);
+
+    var prng = stdx.PRNG.from_seed(bench.seed);
+
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    defer arena.deinit();
+
+    const allocator = arena.allocator();
+
+    const streams = .{
+        try prepare_streams(Values[0], &prng, allocator, streams_count, stream_length),
+        try prepare_streams(Values[1], &prng, allocator, streams_count, stream_length),
+        try prepare_streams(Values[2], &prng, allocator, streams_count, stream_length),
+        try prepare_streams(Values[3], &prng, allocator, streams_count, stream_length),
+        try prepare_streams(Values[4], &prng, allocator, streams_count, stream_length),
+        try prepare_streams(Values[5], &prng, allocator, streams_count, stream_length),
+        try prepare_streams(Values[6], &prng, allocator, streams_count, stream_length),
+    };
+    assert(streams.len == Values.len);
+
+    var checksum: u64 = 0;
+    var duration_samples: [repetitions]stdx.Duration = undefined;
+
+    for (&duration_samples) |*duration| {
+        bench.start();
+        inline for (streams) |pair| {
+            var context = pair.context;
+            context.merge(pair.output);
+            checksum +%= @truncate(pair.output[0].key);
```

Comment:
I am maybe paranoid, but let's use `output[output.len - 1]`, to make sure compiler doesn't stop just after producing the first element. That's mostly magical realism, but better safe than sorry, 🧣 !